main/2024/CG-06.md

```diff
@@ -93,10 +93,1859 @@ scheduled times if previous sessions end early.
+
+YI: For referencing instructions, let’s suppose we solve that, but for other things, hopefully most things don’t need to depend on the binary format that much.
+
+DD: To clarify, for things like @name annotations, are we proposing standardizing that too? 
+
+AR: The name section is already defined in the name section of the standard in an appendix, so this will also go there. I wouldn’t mind moving some of these features into separate documents, otherwise the main document gets cluttered and there’s the question of which features go in the main document versus the other docs. 
+
+TL: Those are the only two concrete annotations that are included in the proposal, the "name" annotation and the "custom" annotation.
+
+DD: If there are annotations that don’t conform to the specific annotations mentioned here, and tooling can add different annotations, it bypasses the standards process to some extent and it would be nice to not do that. It's good to set the precedent to include this in the standard annotations
+
+DS: There’s an important distinction between conventions used by tools and conventions used by engines, in particular engines required to support features in perpetuity. When we get to branch hints, we have to commit to an interpretation in a way that’s more permanent than the linking section.
+
+DD: Mostly want to draw attention to the fact that these could be used for anything, and the tools can then use them - just want to draw attention to this.
+
+Ryan: I don’t think Firefox is going to support any kind of custom section unless it’s required by standards.
+
+AR: There’s a distinction to being standardized, and being in the main document, the branch hinting document will include a separate document, but it will be a standard but also have a different document. In terms of custom sections, they are custom and it is more optional and can be supported optionally
+
+DD: That’s clear. For names, it’s good that the standard is calling it out explicitly. On the tooling side, I think it's good to announce major non-standard custom sections that people are going to use.
+
+AR: In terms of actual custom custom sections that tools use, it would be nice to have some convention to pick names for them to avoid name clobbering. We as a community could define these conventions, but that is orthogonal to the custom sections problem itself.
+
+Dave Thompson (chat): So would the name section be populated by "name" annotations rather than the existing method of extracting names from the symbolic ids of various things (types, funcs, etc.)?
+
+AR: How you construct the Wasm module is up to the tools, there’s no prescriptive ways on how to do it.
+
+**POLL**: Custom annotations to Phase 4
+
+|SF|F|N|A|SA|
+|--|-|-|-|--|
+|14|30|8|0|0|
+
+TL: That is consensus for Phase 4.
+
+Room: *applause*
+
+#### Branch hinting (Yuri Iozzelli)
+
+YI presenting branch hinting (TODO: Add slides)
```

Comment:
@yuri91 do you have a slides link?

Reply:
https://drive.google.com/file/d/12ojxNtRrugs-4MrLqOghCCn2fJbfmqcl/view?usp=sharing